src/MCPServer/tests/test_workflow_abilities.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+import pytest
+
+import workflow
+from workflow_abilities import choice_is_available
+
+
+ASSETS_DIR = os.path.join(
+    os.path.dirname(os.path.abspath(
+        os.path.join(__file__, os.pardir))), "lib", "assets")
+
+
+CREATE_SIP_LINK_ID = "bb194013-597c-4e4a-8493-b36d190f8717"
+SEND_TO_BACKLOG_CHAIN_ID = "7065d256-2f47-4b7d-baec-2c4699626121"
+
+
+@pytest.fixture
+def _workflow():
+    path = os.path.join(ASSETS_DIR, "workflow.json")
+    with open(path) as fp:
+        return workflow.load(fp)
+
+
+def test_choice_is_available__enabled(settings, _workflow):
+    """<Send to backlog> is going to be shown to the user."""
+    settings.SEARCH_ENABLED = ("transfers, aips")
+    link = _workflow.get_link(CREATE_SIP_LINK_ID)
+    chain = _workflow.get_chain(SEND_TO_BACKLOG_CHAIN_ID)
+    assert choice_is_available(link, chain, settings) is True
+
+
+def test_choice_is_available__disabled(settings, _workflow):
+    """<Send to backlog> is not going to be shown to the user."""
+    settings.SEARCH_ENABLED = ("aips")
+    link = _workflow.get_link(CREATE_SIP_LINK_ID)
+    chain = _workflow.get_chain(SEND_TO_BACKLOG_CHAIN_ID)
+    assert choice_is_available(link, chain, settings) is False
```

Comment:
For now I've made `en` mandatory in the schema which actually matches how the reality used to be in our database anyways. This is before we're making our workflow available to users so I'm not too concerned. I'm trying to reduce the scope of this PR as much as possible.

Reply:
Marking as unresolved, I think my fix may not be good enough. Feedback is welcome.

Reply:
To make this work when `fr` is the default language (which I think is what happens when selected through the Dashboard) I had to do:
``` python
from __future__ import unicode_literals


ABILITIES = (
    # The (elastic)search Transfers indexes must be enabled for the transfers
    # backlog. Therefore, the "Send to backlog" choice at the "Create SIP(s)"
    # decision point should not be available when the 'transfers' indexes are
    # not included in the SEARCH_ENABLED setting.
    {
        'name': 'search',
        'enabled_attr': 'SEARCH_ENABLED',
        'enabled_condition': 'in',
        'enabled_value': 'transfers',
        'dependencies': (
            (u'Création de SIP(s)', u'Envoyer au backlog'),
        )
    },
)
```
To guarantee the test passes now, I think we'd need to patch the language variable to `en` like discussed in the other unit tests. 

If English is mandatory in the schema then it does feel like the beginning of one approach but I think a lot hangs on the semantics of what we're doing.

Where modules have access to `workflow.py` but need to do an internal lookup (a non-user facing lookup) then they could specify a `force_lang` attribute, e.g.
``` python
    def get_label(
            self, key, lang=_FALLBACK_LANG, fallback_label=None,
            force_lang=None):
        """Proxy to find translated attributes."""
        if force_lang:
            lang="en"
```
This attribute could be removed in time.

The last few lines of `choice_is_available(...)` would become:
``` python
        if not feature_enabled:
            choice_tuple = (
                link.get_label("description", force_lang="en"),
                chain.get_label("description", force_lang="en"))
            if choice_tuple in ability['dependencies']:
                return False
    return True
```
And perhaps there is something similar that can be done in `dashboard/components/api/views/py` but I haven't considered all the implications yet. 

What do you think?

Reply:
I'm not sure if I see the value of `force_lang`. If you really wanted to be explicit about requiring English, can't you just use `lang="en" What would `force_lang` do differently?

I thought it'd be nice to make the assumption that it's English all the way down. It's like when you add message to a template or you write a log event: it must be in English. And we've taken some extra measurements:

- `_FALLBACK_LANG` is a constant and its value is always "en" regardless user session preferences.
- In `get_label()`, `lang` defaults to `_FALLBACK_LANG`.
- Our schema requires `en` on every translation object.

Reply:
Thanks @sevein, you're right. `lang` would be the right way to do this. If I understand correctly,  `choice_is_available(...)` needs to be updated to request `en` so that it can match the ability's dependencies strings to a string in the `workflow.json` file?

I think I was conflating two issues above. I am still concerned one exists, but my describing a solution doesn't help illustrate that. Let me see if I can describe what I see, and then the problem:

Your  scenario is making sure we can always return 'something'. We'll never have to worry about a string not being returned. If a user requests `hu` (Hungarian), and we haven't got it, they'll get English. And I think that's a good current state for all the strings that we use in Archivematica. And I agree that we've pretty much covered that whenever we're writing data out for the first time.

My problem scenario is reverse-lookup. When we need to lookup data based on a string. I believed that unit test was illustrative of.

The other example is in the API when we set status based on being able to find a job, but the job name is `en`:

```
    ret['microservice'] = job.jobtype
    if job.currentstep == models.Job.STATUS_AWAITING_DECISION:
        ret['status'] = 'USER_INPUT'
    elif 'failed' in job.microservicegroup.lower():
        ret['status'] = 'FAILED'
    elif 'reject' in job.microservicegroup.lower():
        ret['status'] = 'REJECTED'
    elif job.jobtype == 'Remove the processing directory':  # Done storing AIP
        ret['status'] = 'COMPLETE'
    elif models.Job.objects.filter(sipuuid=unit_uuid).filter(jobtype='Create SIP from transfer objects').exists():
        ret['status'] = 'COMPLETE'
        # Get SIP UUID
        sips = models.File.objects.filter(transfer_id=unit_uuid, sip__isnull=False).values('sip').distinct()
        if sips:
            ret['sip_uuid'] = sips[0]['sip']
    elif models.Job.objects.filter(sipuuid=unit_uuid).filter(jobtype='Move transfer to backlog').exists():
        ret['status'] = 'COMPLETE'
        ret['sip_uuid'] = 'BACKLOG'
    else:
        ret['status'] = 'PROCESSING'
```

If I want to return the status as`COMPLETE` by looking for `Create SIP from transfer objects` I can see below that it will only work if the user isn't using any other translation but `en`:

```
            "description": {
                "en": "Create SIP from transfer objects",
                "es": "Crear SIP a partir de los objetos de la transferencia",
                "fr": "Créer un SIP à partir des objets transférés",
                "ja": "転送オブジェクトからSIPを作成する",
                "pt_BR": "Crie SIP a partir de objetos de transferência",
                "sv": "Skapa SIP från överföringspaketets objekt"
```

I feel like we need to identify anywhere in Archivematica where we want to compare something to a workflow string written in English and understand what a solution to that lookup looks like, whether that is setting `lang='en'` or something else, e.g. in the API case. 

I guess I'm trying to work through in my mind what a solution looks like now until we're not relying on strings like this. 

But IDK, does this make sense?

Reply:
I see what you mean but I thought we've confirmed that `jobChainLink` always uses English to persist the string in `Job.jobtype` precisely to ensure that we were not breaking code like the one you've described above.

Reply:
You're absolutely right! I must have got something wrong in my testing as I was seeing a failure before. I have just verified this with the API and in the automation tools as well. I think we're looking good. Will touch-base in Slack! :+1: